datafusion/core/src/datasource/file_format/file_type.rs

```diff
@@ -162,8 +173,13 @@ impl FileCompressionType {
             BZIP2 => Box::new(BzDecoder::new(r)),
             #[cfg(feature = "compression")]
             XZ => Box::new(XzDecoder::new(r)),
+            #[cfg(feature = "compression")]
+            ZSTD => match ZstdDecoder::new(r) {
+                Ok(decoder) => Box::new(decoder),
+                Err(e) => return Err(DataFusionError::External(Box::new(e))),
```

Comment:
I'm a little confused about why we use `DataFusionError::External` warp the error instead of return directly.